nav2-usability/scenario-code/skipping-stacks/router.dart

```diff
@@ -0,0 +1,503 @@
+// Copyright 2021, the Flutter project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+
+final List<Book> books = [
+  Book('Stranger in a Strange Land', Author('Robert A. Heinlein')),
+  Book('Foundation', Author('Isaac Asimov')),
+  Book('Fahrenheit 451', Author('Ray Bradbury')),
+];
+
+List<Author> get authors => [
+  ...books.map((book) => book.author),
+];
+
+void main() {
+  runApp(BooksApp());
+}
+
+class Book {
+  final String title;
+  final Author author;
+
+  Book(this.title, this.author);
+}
+
+class Author {
+  String name;
+
+  Author(this.name);
+}
+
+class AppState extends ChangeNotifier {
+  bool _showAuthorsPage = false;
+  int? _selectedBookId;
+  int? _selectedAuthorId;
+
+  int? get selectedBookId => _selectedBookId;
```

Comment:
We should group setter and getter together
```
int? get selectedBookId => _selectedBookId;
int? _selectedBookId;
set selectedBookId(int? id) {
    if (id == null || id < 0 || id > books.length - 1) {
      return;
    }

    _showAuthorsPage = false;
    _selectedBookId = id;
    _selectedAuthorId = null;
    notifyListeners();
  }
```
same for `selectedAuthorId`

Reply:
as well as `showAuthorsPage`

Reply:
I decided to group the getters and setters, but left the fields at the top of the class.